toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1355,23 +1449,33 @@ func (g *PkgGraph) fixPrebuiltSRPMsCycle(trimmedCycle []*PkgNode) (err error) {
 
 			logger.Log.Debugf("Adding a 'PreBuilt' node '%s' with id %d.", preBuiltNode.FriendlyName(), preBuiltNode.ID())
 
-			parentNodes := g.To(currentNode.ID())
-			for parentNodes.Next() {
-				parentNode := parentNodes.Node().(*PkgNode)
-				if parentNode.Type == TypeLocalBuild && parentNode.SrpmPath == previousNode.SrpmPath {
-					g.RemoveEdge(parentNode.ID(), currentNode.ID())
-
-					err = g.AddEdge(parentNode, preBuiltNode)
-					if err != nil {
-						logger.Log.Errorf("Adding edge failed for %v -> %v", parentNode, preBuiltNode)
-						return
-					}
-				}
-			}
+			g.replaceCurrentNodeWithPrebuiltNode(currentNode, preBuiltNode, previousNode)
+			logger.Log.Infof("Cycle fixed using prebuilt node: %s with id %d", preBuiltNode.FriendlyName(), preBuiltNode.ID())
 
 			return
 		}
-
+		if resolveCyclesFromUpstream {
```

Comment:
Do we want to check `buildToRunEdge` here as well?

Reply:
was checking buildToRunEdge before modifying the graph. Now, moved the check above all cycle resolution.